src/hypertable.c

```diff
@@ -849,6 +853,160 @@ hypertable_create_schema(const char *schema_name)
 		);
 }
 
+#define INSERT_BLOCKER_NAME "insert_blocker"
+
+TS_FUNCTION_INFO_V1(hypertable_insert_blocker);
+
+Datum
+hypertable_insert_blocker(PG_FUNCTION_ARGS)
+{
+	TriggerData *trigdata = (TriggerData *) fcinfo->context;
+	const char *relname = get_rel_name(trigdata->tg_relation->rd_id);
+
+	if (!CALLED_AS_TRIGGER(fcinfo))
+		elog(ERROR, "insert_blocker: not called by trigger manager");
+
+	ereport(ERROR,
```

Comment:
Let's throw a different error  if restoring is on vs any other error? That way will be easier to debug with clients.

Reply:
yeah, that's a good idea.